backend/src/services/userService.ts

```diff
@@ -112,13 +124,133 @@ const createUser = async (
     }
 };
 
+/**
+ * Fetches and returns list of projects by user id.
+ * returns null on failure
+ */
+const getProjects = async (
+    id: string
+): Promise<{ id: string; name: string }[] | null> => {
+    try {
+        const projects = await selectProjectsbyUserID(id);
+        return projects;
+    } catch (e) {
+        logger.error(e);
+        return null;
+    }
+};
+
+/**
+ * Fetches data of a specific project
+ * along with a boolean value, which is
+ * true if successful or false
+ * if project does not exist or user id
+ * does not match or on failure
+ */
+const getProject = async (
+    projectID: string,
+    userID: string
+): Promise<[boolean, { data: object }]> => {
```

Comment:
I think instead of `[boolean, { data: object }]`
this could be 
```ts
{ success: boolean, data: object }
```
to make the return value more explicit (`if (res.success) { ... }` instead of `if (res[0]) { ... }`)